Sources/SKTestSupport/TestSourceKitLSPClient.swift

```diff
@@ -72,7 +72,7 @@ public final class TestSourceKitLSPClient: MessageHandler {
   ///
   /// `isOneShort` if the request handler should only serve a single request and should be removed from
   /// `requestHandlers` after it has been called.
-  private nonisolated(unsafe) var requestHandlers: ThreadSafeBox<[(requestHandler: Any, isOneShot: Bool)]> =
+  private nonisolated(unsafe) var requestHandlers: ThreadSafeBox<[(requestHandler: Sendable, isOneShot: Bool)]> =
```

Comment:
One thing I've just noticed here is that when we're doing `requestHandlers.value.append`, there's a gap between where we read the array and write back the array where e.g the request handler could remove one of the handlers, but then we'd immediately write it back

Reply:
Oh, good catch. It’s only test code, so a little bit of 🤷🏽 but I changed them to use `withLock`. Do you have an idea if there’s a way to prevent this in the type system? I couldn’t come up with anything but maybe you have ideas. If not: Whatever, should be fine.

Reply:
I _think_ using `_modify` should allow it to be mutated while covered by the lock.

Reply:
Wohooo. That seems to have worked.